pkg/sources/gitlab/gitlab.go

```diff
@@ -590,6 +586,51 @@ func normalizeRepos(repos []string) ([]string, []error) {
 	return validRepos, errs
 }
 
+// normalizeGitlabEndpoint ensures that if and endpoint is going to gitlab.com, we use https://gitlab.com/ as the endpoint.
+// If we see the protocol is http, we error, because this shouldn't be used.
+// Otherwise, it ensures we are using https as our protocol, if none was provided.
+func normalizeGitlabEndpoint(gitlabEndpoint string) (string, error) {
+	if gitlabEndpoint == "" {
+		return cloudBaseURL, nil
+	}
+
+	gitlabURL, err := url.Parse(gitlabEndpoint)
+	if err != nil {
+		return "", err
+	}
+
+	// We probably didn't receive a URL with a scheme, which messed up the parsing.
+	if gitlabURL.Host == "" {
+		gitlabURL, err = url.Parse("https://" + gitlabEndpoint)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return normalizeGitlabEndpointURL(gitlabURL)
+}
+
+// Do not use this, only use normalizeGitlabEndpoint.
```

Comment:
getting your entry for the "most ignored comment twelve months from now" competition ready?

i'm guessing you just wanted the `normalizeGitlabEndpointURL` modular but not in a way that causes anyone else to consume it as such. this comment prompted a double-take in me because it looks way scarier than that - i'd vote for just inlining this function (unless i'm misunderstanding something)

Reply:
Yeah, inlining is definitely better than a scary comment. Also it was initially done this way to more or less allow the overall function to be recursive on the "try again but with https:// at the start" without needing to worry that we'd ever hit that case again.

Reply:
ah ok that makes sense